### DISCUSSION TITLE
Select condition on unnest over an array of struct

### DISCUSSION THREAD

**Discussion by aczire** [sqlalchemy/sqlalchemy#10779] [Usage Questions]
I was trying to select rows from a BigQuery view based on a condition which is inside an array of structures,
Below is the SQL query

```
        SELECT id, enId, versionId, code_coding
        FROM `some.schema_views.table`, UNNEST(code_coding) AS cc
        WHERE (cc.code, cc.display) = ('H', 'BLAST')
        LIMIT 100
```
Below is my failed attempt to write the same in SQLAlchemy

```
        from sqlalchemy.sql.expression import cast
        from sqlalchemy import String, func
        code_coding = db.query(func.unnest(self.model.code_coding)).subquery()
        conditions.append(
            and_(cast(code_coding.c.code, String) == "H", cast(code_coding.c.display, String) == "BLAST")
        )
```
It says `c.code` is not found. Any idea? Thanks ahead!

Using BigQuery

**Comment by CaselIT** [comment]
Hi,

to generate an unnest in the from clause you would use https://docs.sqlalchemy.org/en/20/core/functions.html#sqlalchemy.sql.functions.FunctionElement.table_valued not `code_coding = db.query(func.unnest(self.model.code_coding)).subquery()`
